src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -466,13 +464,9 @@ public void saveProgram(File f) throws IOException {
             throw new IOException("Failed to persist "+f,e);
         } finally {
             PROGRAM_STATE_SERIALIZATION.set(old);
+            execution.time(TimingKind.save, true);
             Util.deleteFile(tmpFile);
         }
-
-        if (logging) {
```

Comment:
Yeah, I would say _if_ timing information suggests that program saving is a major bottleneck then we go back and see why.